src/main/scala/ru/d4team/chat/services/ChatRoomsService.scala

```diff
@@ -42,4 +43,11 @@ final case class ChatRoomsServiceMapImpl(private[services] val rooms: ChatRoomsM
         case None       => None // TODO (Aleksei Litkovetc) - Log a leaving from non-existent room
```

Comment:
it might be better to give names using functions for example

case Some(room) => JoinToNonEmptyRoom
case None => createRoomAndJoin
of course if I understood logic correctly)

Reply:
As for me, so highly grained decomposition (with one line functions) won't bring a benefit. It's not complex to read logic like this `room.addOne(channel.id, channel).some`.

If you don't mind, I would leave it as is.